### ISSUE TITLE
[Bug]: CJK glyphs in GTK applications rendering as "glyph not found" ("tofu") characters

### ISSUE THREAD

**Issue by ssokolow** [flatpak/flatpak#4641]
### Checklist

- [X] I agree to follow the [Code of Conduct](https://github.com/flatpak/flatpak/blob/master/CODE_OF_CONDUCT.md) that this project adheres to.
- [X] I have searched the [issue tracker](https://www.github.com/flatpak/flatpak/issues) for a bug that matches the one I want to file, without success.
- [X] If this is an issue with a particular app, I have tried filing it in the appropriate issue tracker for the app (e.g. under https://github.com/flathub/) and determined that it is an issue with Flatpak itself.

### Flatpak version

1.11.2

### What Linux distribution are you using?

Ubuntu

### Linux distribution version

Kubuntu 20.04.3 LTS

### What architecture are you using?

x86_64

### How to reproduce

1. Install a GTK-based application through Flatpak such as [Bustle](https://github.com/flathub/org.freedesktop.Bustle/issues/10), [Deluge](https://github.com/flathub/org.deluge_torrent.deluge/issues/6), [Flatseal](https://github.com/flathub/com.github.tchx84.Flatseal/issues/42), [HandBrake](https://github.com/HandBrake/HandBrake/issues/4056), [Inkscape](https://github.com/flathub/org.inkscape.Inkscape/issues/100), or jstest-gtk.
2. If necessary to reproduce (uncertain), set `filesystem=!host` and/or `filesystem=!home` overrides.
3. Copy-paste a string of CJK text, such as "文字化け" into a text-entry field within the application

### Expected Behavior

The application should render it correctly, like GTK applications not installed through Flatpak or Qt, Java, and custom-toolkitted apps operating under the same Flatpak restrictions.

Tested-working examples include:

**Qt:**
* `io.github.antimicrox.antimicrox`
* `org.gottcode.FocusWriter`
* `org.kde.kcachegrind`
* `org.kde.krita`
* `io.lmms.LMMS`
* `nz.mega.MEGAsync`
* `org.kde.okteta`
* `org.zealdocs.Zeal`

**Java:**
* `cc.arduino.arduinoide`
* `org.jdownloader.JDownloader`
* `com.sweethome3d.Sweethome3d`

**Custom:**
* `org.blender.Blender`
* `org.mozilla.firefox`
* `org.godotengine.Godot`
* `org.mozilla.Thunderbird`
* `com.github.micahflee.torbrowser-launcher`
* `com.github.Eloston.UngoogledChromium`

(For example, AntiMicroX renders CJK fonts perfectly well with filesystem access limited to `xdg-config/kdeglobals:ro` and Arduino IDE renders CJK fine with only `~/src/arduino:create`.)

### Actual Behavior

CJK glyphs render as ["tofu"](https://en.wikipedia.org/wiki/Specials_(Unicode_block)) characters. (i.e. Windows-style empty rectangle renderings of the `.notdef` glyph)

**Comment by ssokolow** [comment]
While answering some questions for the author of Bustle, I noticed some useful CLI output:

```
(bustle:2): Pango-WARNING **: 06:20:58.331: failed to create cairo scaled font, expect ugly output. the offending font is 'VL Gothic 9.9990234375'

(bustle:2): Pango-WARNING **: 06:20:58.332: font_face status is: file not found

(bustle:2): Pango-WARNING **: 06:20:58.332: scaled_font status is: file not found

(bustle:2): Pango-WARNING **: 06:20:58.349: failed to create cairo scaled font, expect ugly output. the offending font is 'VL Gothic 9.9990234375'

(bustle:2): Pango-WARNING **: 06:20:58.349: font_face status is: file not found

(bustle:2): Pango-WARNING **: 06:20:58.349: scaled_font status is: file not found
```

If it's the Pango in `org.freedesktop.Platform` that's going awry, while Qt, Java, and things like Firefox and Chromium aren't using it, that'd certainly explain what's going on.

Now to figure out why I'm experiencing the problem and Bustle's author isn't.

**Comment by ssokolow** [comment]
`sudo apt-get remove fonts-vlgothic` makes it work, so it looks like there's something about the VL Gothic font that causes Pango in the Flatpak runtime to bomb out, but not in the host OS.

**Comment by ssokolow** [comment]
I get the following output (and a re-broken Flatpak Pango) when reinstalling it, so, if it *doesn't* break on your system with VL Gothic installed, I'd check where it is in the fallback chain:

```
update-alternatives: using /usr/share/fonts/truetype/vlgothic/VL-Gothic-Regular.ttf to provide /usr/share/fonts/truetype/fonts-japanese-gothic.ttf (fonts-japanese-gothic.ttf) in auto mode
```

**Comment by luzpaz** [comment]
Any progress here ?

**Comment by Alerymin** [comment]
Similar issue here

Here is a comparison between a folder displayed from Wine in Bottles flatpak, and one from Wine in Lutris as a system app.
Both have cjkfonts installed and the flatpak has ja_JP.UTF-8 as extra language

Some names are only partially displayed or even not at all

![Image](https://github.com/user-attachments/assets/7cc49bc8-bcb1-4d6b-8cac-2f5dbdbb22b2)

![Image](https://github.com/user-attachments/assets/e28d3f4a-8c44-49c4-b8b1-3794a8211616)

**Comment by dnet890** [comment]
Yeah I use Kubuntu and Nicotine and Quod Libet have same issues as well.
